src/web/client/test/integration/schemaHelperUtil.test.ts

```diff
@@ -0,0 +1,791 @@
+ */
+
+import { expect } from "chai";
+import sinon from "sinon";
+import {
+    entityAttributesWithBase64Encoding,
+    schemaEntityName,
+    schemaKey,
+} from "../../schema/constants";
+import * as portalSchemaReader from "../../schema/portalSchemaReader";
+import {
+    getPortalSchema,
+    getEntity,
+    getAttributePath,
+    isBase64Encoded,
+    encodeAsBase64,
+    isWebFileV2,
+    getLanguageIdCodeMap,
+    getWebsiteIdToLanguageMap,
+    getwebsiteLanguageIdToPortalLanguageMap,
+} from "../../utilities/schemaHelperUtil";
+import {
+    portal_schema_V1,
+    portal_schema_V2,
+} from "../integration/portalSchemaReader.mock";
+import WebExtensionContext from "../../WebExtensionContext";
+import * as Constants from "../../common/constants";
+
+describe("schemaHelperUtil_ShouldReturnportalSchemaV2", () => {
+    afterEach(() => {
+        sinon.restore();
+    });
+    it("getPortalSchema_ShouldReturnportalSchemaV2", () => {
+        //Act
+        const schema = "portalschemav2";
+        //Action
+        const result = getPortalSchema(schema);
+        //Assert
+
+        expect(result).deep.equal(portal_schema_V2);
+    });
+
+    it("getPortalSchema_ShouldReturnportalSchemaV2IfSchemaIsInCaps", () => {
+        //Act
+        const schema = "portalschemav2".toUpperCase();
+        //Action
+        const result = getPortalSchema(schema);
+        //Assert
+
+        expect(result).deep.equal(portal_schema_V2);
+    });
+
+    it("getPortalSchema_ShouldReturnportalSchemaV1IfSchemaDoNotMatch", () => {
+        //Act
+        const schema = "portalschemaV1";
+        //Action
+        const result = getPortalSchema(schema);
+        //Assert
+
+        expect(result).deep.equal(portal_schema_V1);
+    });
+
+    it("getEntity_ShouldReturnEntitiesSchemaMapWhenSchemaIsportalschemav2", () => {
+        //Act
+        const schema = "portalschemav2";
+        const entity = "webPages";
+        const queryParamsMap = new Map<string, string>([
+            [schemaKey.SCHEMA_VERSION, schema],
+        ]);
+
+        const entityMap = new Map<string, string>([["test", "Entity"]]);
+
+        const entitiesSchemaMap = new Map<string, Map<string, string>>([
+            [entity, entityMap],
+        ]);
+
+        sinon
+            .stub(portalSchemaReader, "getDataSourcePropertiesMap")
+            .returns(
+                new Map<string, string>([
+                    [schemaKey.SINGLE_ENTITY_URL, schemaKey.SINGLE_ENTITY_URL],
+                ])
+            );
+
+        sinon
+            .stub(portalSchemaReader, "getEntitiesSchemaMap")
+            .returns(entitiesSchemaMap);
+        WebExtensionContext.setWebExtensionContext("", "", queryParamsMap);
+        //Action
+        const result = getEntity(entity);
+        //Assert
+        expect(result).deep.be.eq(entityMap);
+    });
+
+    it("getEntity_ShouldReturnEntitiesSchemaMapWhenSchemaIsportalschemav2AndInCaps", () => {
+        //Act
```

Comment:
This test case looks redundant because we are always changing the schemaName in Uppercase

Reply:
I wrote this scenario because in the called function toLowerCase() is used 

![image](https://user-images.githubusercontent.com/114223231/212885612-eb90783c-2fe0-4824-9f53-dff9236243a7.png)

Reply:
Thats fine. We dont't need that much coverage too. 100% coverage means we are overdoing

Reply:
Removing